### PR TITLE
Data type fixes

### PR DESCRIPTION
**Pull request by sharadregoti**
1) Now you can specify tablename.columnname syntax in mongodb where clause. space cloud will remove the tablename prefix depending upon the collection being queried.
2) In & not in operator now works with boolean data type
3) Now we can use logical operators on datetime data type in mongodb
4) Mysql will now use decimal data type for float you can specify the precision & scale in the schema
5) Fixed DateTime data type in mysql not storing milli/nano seconds.
6) Fixed Time data type not storing mill/nano seconds

### CODE REVIEW

File: gateway/model/schema_type.go [hunk 1]
```diff
@@ -15,6 +15,12 @@ type (
 		IsFieldTypeRequired bool   `json:"isFieldTypeRequired"`
 		IsList              bool   `json:"isList"`
 		Kind                string `json:"kind"`
+		// Precision is used to hold precision information for data types Float
+		// It represent number the digits to be stored
+		Precision int `json:"precision"`
```
Comment (YourTechBud) [inline gateway/model/schema_type.go:20]: Store such things inside an args struct

File: gateway/modules/crud/mgo/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		args args
+		want map[string]interface{}
+	}{
+		{
+			name: "Where clause with collection name separated by dot notation",
+			args: args{
+				ctx: context.Background(),
+				col: "users",
+				find: map[string]interface{}{
+					"users.id":         1,
```
Comment (YourTechBud) [inline gateway/modules/crud/mgo/helpers_test.go:26]: Add one more field, users.address.street. the result should be address.street

File: gateway/modules/crud/mgo/read.go [hunk 3]
```diff
@@ -24,6 +24,8 @@ func (m *Mongo) Read(ctx context.Context, col string, req *model.ReadRequest) (i
 	}
 	collection := m.getClient().Database(m.dbName).Collection(col)
 
+	req.Find = sanitizeWhereClause(ctx, col, req.Find)
```
Comment (YourTechBud) [inline gateway/modules/crud/mgo/read.go:27]: Are you doing this for update and delete as well?

**Review by YourTechBud — CHANGES_REQUESTED** [review]

File: gateway/modules/crud/mgo/read.go (hunk 3 above)
Comment (sharadregoti) [reply gateway/modules/crud/mgo/read.go:27]: Nope, I'll add it

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**